File: lab4/s1/web/src/main/java/ru/uvuv643/web/EnumValueValidator.java
package ru.uvuv643.web;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;
import java.util.ArrayList;
import java.util.List;

public class EnumValueValidator implements ConstraintValidator<EnumValue, String> {
    private List<String> acceptedValues = new ArrayList<>();

    @Override
    public void initialize(EnumValue constraintAnnotation) {
        for (Enum<?> enumValue : constraintAnnotation.enumClass().getEnumConstants()) {
            acceptedValues.add(enumValue.name());
        }
    }

    @Override
    public boolean isValid(String value, ConstraintValidatorContext context) {
        return value == null || acceptedValues.contains(value);
    }
}

File: lab2/s1/src/main/kotlin/ru/uvuv643/soa/server/EnumValueValidator.kt
package ru.uvuv643.soa.server

import EnumValue
import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

class EnumValueValidator : ConstraintValidator<EnumValue, String> {
    private val acceptedValues: MutableList<String> = mutableListOf()

    override fun initialize(constraintAnnotation: EnumValue) {
        super.initialize(constraintAnnotation)
        acceptedValues.addAll(constraintAnnotation.enumClass.java.enumConstants.map { it.name })
    }

    override fun isValid(value: String?, context: ConstraintValidatorContext?): Boolean {
        return if (value == null) {
            true
        } else acceptedValues.contains(value.toString())
    }
}